test/test-annotation.js

```diff
@@ -280,6 +285,30 @@ describe('Annotations', function () {
         done();
       });
     });
+
+    it('should delete annotation with no replies', function () {
+      passportStub.login(user);
+      chai.request(app)
+      .delete(`/api/annotation/${StupidAnnotation.id}`)
+      .end(function (err, res) {
+        res.should.have.status(200);
+        res.body.should.have.property('SUCCESS');
+        // TODO: when check db here, annotation doesn't exist so whats up?
+      });
+
+      return util.checkDatabase(resolve => {
+        resolve(Annotation.findById(StupidAnnotation.id).should.eventually.be.null);
+      });
+    });
+
+    it('should delete annotation with reply', function () {
+      passportStub.login(user);
+      chai.request(app)
+        .delete(`/api/annotation/${PublicAnnotation.id}`)
+        .end(function (err, res) {
+          res.should.have.status(200);
+        });
+    });
```

Comment:
Should add checks on returned documents and database contents for both of these tests.